### ISSUE TITLE
Rule proposal: `prefer-node-remove`

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-unicorn#213]
Prefer [`node.remove()`](https://developer.mozilla.org/en-US/docs/Web/API/ChildNode/remove) over [`node.parentNode.removeChild(node)`](https://developer.mozilla.org/en-US/docs/Web/API/Node/removeChild). `removeChild` is the legacy way of doing it.

## Fail

```js
foo.parentNode.removeChild(foo);
this.parentNode.removeChild(this);
```

## Pass

```js
foo.remove();
this.remove();
```

**Comment by IssuehuntBot** [comment]
@issuehuntfest has funded $60.00 to this issue. [See it on IssueHunt](https://issuehunt.io/repos/55832243/issues/213)

**Comment by medusalix** [comment]
Working on it 👍.

**Comment by IssuehuntBot** [comment]
@doniyor2109 has submitted a pull request. [See it on IssueHunt](https://issuehunt.io/repos/55832243/issues/213)

**Comment by doniyor2109** [comment]
@medusalix I did not see that you are working on it. I will close my PR as I did not notify about it. Sorry

**Comment by medusalix** [comment]
Oh, I'm so sorry that you've already put time into the PR. Thank you for your kindness 👍.

**Comment by IssuehuntBot** [comment]
@sindresorhus has rewarded $54.00 to @medusalix. [See it on IssueHunt](https://issuehunt.io/repos/55832243/issues/213)

- :moneybag: Total deposit: $60.00
- :tada: Repository reward(0%): $0.00
- :wrench: Service fee(10%): $6.00
